server/src/main/java/org/elasticsearch/index/analysis/PreBuiltAnalyzerProviderFactory.java

```diff
@@ -46,13 +47,17 @@
     PreBuiltAnalyzerProviderFactory(String name, PreBuiltAnalyzers preBuiltAnalyzer) {
         super(name, new PreBuiltAnalyzersDelegateCache(name, preBuiltAnalyzer));
         this.create = preBuiltAnalyzer::getAnalyzer;
-        current = new PreBuiltAnalyzerProvider(name, AnalyzerScope.INDICES, preBuiltAnalyzer.getAnalyzer(Version.CURRENT));
+        Analyzer analyzer = preBuiltAnalyzer.getAnalyzer(Version.CURRENT);
+        analyzer.setVersion(Version.CURRENT.luceneVersion);
+        current = new PreBuiltAnalyzerProvider(name, AnalyzerScope.INDICES, analyzer);
     }
 
-    public PreBuiltAnalyzerProviderFactory(String name, PreBuiltCacheFactory.CachingStrategy cache, Function<Version, Analyzer> create) {
+    public PreBuiltAnalyzerProviderFactory(String name, PreBuiltCacheFactory.CachingStrategy cache, Supplier<Analyzer> create) {
         super(name, cache);
-        this.create = create;
-        this.current = new PreBuiltAnalyzerProvider(name, AnalyzerScope.INDICES, create.apply(Version.CURRENT));
+        this.create = version -> create.get();
```

Comment:
I think you can change this to a `Supplier<Analyzer>` now.

Reply:
It is still needed at line 49, otherwise we can't delegate the request version all the down to `PreBuiltAnalyzers#getAnalyzer(...)`. Once all analyzers have been ported then I will change the type of the create field to `Supplier<Analyzer>`.

Reply:
Ah! I get it now. LGTM